stix2validator/output.py

```diff
@@ -142,50 +141,72 @@ def print_horizontal_rule():
+    """Print the results of validating an object.
+
+    Args:
+        obj_result: An ObjectValidationResults instance.
+
+    """
+    print_results_header(obj_result.object_id, obj_result.is_valid)
+
+    if obj_result.warnings:
+        print_warning_results(obj_result, 1)
+    if obj_result.errors:
+        print_schema_results(obj_result, 1)
+
+
+def print_file_results(file_result):
+    """Print the results of validating a file.
+
+    Args:
+        file_result: A FileValidationResults instance.
+
+    """
+    print_results_header(file_result.filepath, file_result.is_valid)
+
+    for object_result in file_result.object_results:
+        if object_result.warnings:
+            print_warning_results(object_result, 1)
+        if object_result.errors:
+            print_schema_results(object_result, 1)
+
+    if file_result.fatal:
+        print_fatal_results(file_result.fatal, 1)
+
+
 def print_results(results):
-    """Prints `results` (the results of validation) to stdout.
+    """Print `results` (the results of validation) to stdout.
 
     Args:
-        results: A list of FileValidationResults instances.
+        results: A list of FileValidationResults or ObjectValidationResults
+                 instances.
 
     """
     if not isinstance(results, list):
         results = [results]
 
-    if results[0].__class__.__name__ is 'FileValidationResults':
-        for file_result in sorted(results, key=operator.attrgetter("filepath")):
-            print_horizontal_rule()
-            print_level(logger.info, "[-] Results for: %s", 0, file_result.filepath)
-
-            if file_result.is_valid:
-                marker = _GREEN + "[+]"
-                verdict = "Valid"
-                log_func = logger.info
-            else:
-                marker = _RED + "[X]"
-                verdict = "Invalid"
-                log_func = logger.error
-            print_level(log_func, "%s STIX JSON: %s", 0, marker, verdict)
-
-            for object_result in file_result.object_results:
-                if object_result.warnings:
-                    print_warning_results(object_result, 1)
-                if object_result.errors:
-                    print_schema_results(object_result, 1)
-
-            if file_result.fatal:
-                print_fatal_results(file_result.fatal, 1)
-    elif results[0].__class__.__name__ is 'ObjectValidationResults':
-        valid = True
-        for obj_result in results:
-            if obj_result.warnings:
-                print_warning_results(obj_result)
-                valid = False
-            if obj_result.errors:
-                print_schema_results(obj_result)
-                valid = False
-        if valid:
-            print_level(logger.info, "STIX JSON: Valid", 0)
-    else:
-        raise ValueError('Argument to print_results() must be a list of '
-                         'FileValidationResults or a list of ObjectValidationResults.')
+    for r in results:
```

Comment:
This loses the ability to print a list of `FileValidationResults` in sorted order by filepath, but I think that's OK.